### ISSUE TITLE
UTF-8 Encode/Decode Error Handling

### ISSUE THREAD

**Issue by ashurack** [splunk/splunk-sdk-python#505]
**Describe the bug**
Custom search commands exception out when non UTF-8 event data is present in the search pipeline

**To Reproduce**
1. Create a custom command
2. Pass non UTF-8 field data to the custom command (feel free to use [invalid_utf8.csv](https://github.com/splunk/splunk-sdk-python/files/10396930/invalid_utf8.csv))

**Expected behavior**
splunk-sdk-python (and all other potentially impacted SDK's) should handle encoding/decoding in the same manner as Splunk Core.

**Logs or Screenshots**
* Not working
[broken_search.log](https://github.com/splunk/splunk-sdk-python/files/10396806/broken_search.log)
![image](https://user-images.githubusercontent.com/70148548/211943163-35d1ef85-c584-4bb3-8428-d7ef4e9e0000.png)

* After patching six.py
![image](https://user-images.githubusercontent.com/70148548/211944171-2a125dfa-fc5f-4490-a152-80e55f31288d.png)

**Splunk (please complete the following information):**
- Version: 8.2.5
- OS: Windows 10 Pro 19045.2486
- Deployment: single-instance

**SDK (please complete the following information):**
 - Version: 1.7.2
 - Language Runtime Version: Python 3.7
 - OS: Windows 10 Pro 19045.2486

**Additional context**
My patch - to get my command working ASAP - was to change `errors='strict'` to `errors='replace'` [here](https://github.com/splunk/splunk-sdk-python/blob/1.7.2/splunklib/six.py#L901).  I chose [replace](https://docs.python.org/3/library/codecs.html#error-handlers) since it mimic's the functionality of Splunk.  I didn't touch any other instances of `errors='strict'` and only tested this against `StreamingCommand`.

This bug is not limited to the `inputlookup` command but it is the easiest way to reproduce.

**Comment by ashah-splunk** [comment]
@ashurack we are unable to reproduce the issue and were able to successfully upload the given csv file. below is the screenshot for the same
<img width="1786" alt="image" src="https://user-images.githubusercontent.com/84129413/213102661-e25cdf71-7391-482e-b661-9ff741cc3316.png">

Note- here we are using a [Streaming CSC](https://github.com/splunk/splunk-app-examples/tree/master/custom_search_commands/python/streamingsearchcommands_app). No modification is being applied to the data read from the csv file.

Request you to share your CSC if possible. Also do let us know if something is being missed from steps to reproducing the issue

**Comment by ashurack** [comment]
Looks like the csv file I uploaded is 100% valid UTF.  I'll try to get a sample that will trigger the decoding issue this week.  Message me on Splunk Slack in the meantime for more details.

**Comment by pabloperezj** [comment]
Hi @ashah-splunk, same error here (Splunk 9.0.1, Debian GNU/Linux 11, Python 3.7.11). Loading events from CSV doesn't work as expected (non UTF characters are parsed to UTF). I am using [botsv3 dataset](https://github.com/splunk/botsv3) and getting the same problem as @ashurack using `vt4splunk` streaming command from [VT4Splunk](https://splunkbase.splunk.com/app/6654). The suggested solution by @ashurack seems to work properly.

- Event example:
<img width="964" alt="Screenshot 2023-06-14 at 10 38 36" src="https://github.com/splunk/splunk-sdk-python/assets/122302023/393c8fd8-da22-4d19-8f4b-7d92b2998ea0">

- Error:
<img width="804" alt="Screenshot 2023-06-14 at 14 36 09" src="https://github.com/splunk/splunk-sdk-python/assets/122302023/63cc4882-4c87-47e0-87b3-2de0dde80774">

- search.log:
```
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr: UnicodeDecodeError at "/opt/splunk/etc/apps/TA-virustotal-app/bin/ta_virustotal_app/aob_py3/splunklib/six.py", line 917 : 'utf-8' codec can't decode byte 0x8e in position 294568: invalid start byte
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr: Traceback:
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr:   File "/opt/splunk/etc/apps/TA-virustotal-app/bin/ta_virustotal_app/aob_py3/splunklib/searchcommands/search_command.py", line 780, in _process_protocol_v2
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr:     self._execute(ifile, None)
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr:   File "/opt/splunk/etc/apps/TA-virustotal-app/bin/ta_virustotal_app/aob_py3/splunklib/searchcommands/streaming_command.py", line 55, in _execute
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr:     SearchCommand._execute(self, ifile, self.stream)
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr:   File "/opt/splunk/etc/apps/TA-virustotal-app/bin/ta_virustotal_app/aob_py3/splunklib/searchcommands/search_command.py", line 855, in _execute
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr:     self._execute_v2(ifile, process)
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr:   File "/opt/splunk/etc/apps/TA-virustotal-app/bin/ta_virustotal_app/aob_py3/splunklib/searchcommands/search_command.py", line 948, in _execute_v2
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr:     result = self._read_chunk(istream)
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr:   File "/opt/splunk/etc/apps/TA-virustotal-app/bin/ta_virustotal_app/aob_py3/splunklib/searchcommands/search_command.py", line 912, in _read_chunk
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr:     return metadata, six.ensure_str(body)
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr:   File "/opt/splunk/etc/apps/TA-virustotal-app/bin/ta_virustotal_app/aob_py3/splunklib/six.py", line 917, in ensure_str
ERROR ChunkedExternProcessor [1549763 ChunkedExternProcessorStderrLogger] - stderr:     s = s.decode(encoding, errors)
ERROR ChunkedExternProcessor [1549758 localCollectorThread] - EOF while attempting to read transport header read_size=0
ERROR ChunkedExternProcessor [1549758 localCollectorThread] - Error in 'vt4splunk' command: External search command exited unexpectedly with non-zero error code 1.
ERROR LocalCollector [1549758 localCollectorThread] - SearchMessage orig_component=LocalCollector sid= message_key= message=Error in 'vt4splunk' command: External search command exited unexpectedly with non-zero error code 1.
```

**Comment by ashah-splunk** [comment]
@pabloperezj sorry for the delay in response. We were able to reproduce the issue using [botsv3 dataset](https://github.com/splunk/botsv3).  Also during our verification we found that issue occurs only for certain specific non-utf8 characters. We are validating the change suggested by @ashurack and accordingly will make the changes in the SDK.

We will update you know once we have a new SDK release available with the change.
